src/Simulation/Simulation.jl

```diff
@@ -829,13 +829,17 @@ function _calculate_potential!( sim::Simulation{T, CS}, potential_type::UnionAll
         paint_contacts::Bool = true,
         verbose::Bool = true,
         device_array_type::Type{<:AbstractArray} = Array,
+        initialize::Bool = true
     )::Nothing where {T <: SSDFloat, CS <: AbstractCoordinateSystem}
 
     begin # preperations
         onCPU = !(device_array_type <: GPUArrays.AnyGPUArray)
         convergence_limit::T = T(convergence_limit)
         isEP::Bool = potential_type == ElectricPotential
         isWP::Bool = !isEP
+        if !initialize
+            grid = isEP ? sim.electric_potential.grid : sim.weighting_potentials[contact_id].grid
+        end
```

Comment:
How about you put this as default in line 817? Instead of
```julia
grid::Union{Missing, Grid{T}} = missing
```
put something like
```julia
grid::Union{Missing, Grid{T}} = initialize ? missing : (isEP ? sim.electric_potential.grid : sim.weighting_potential[contact_id].grid)
```